### ISSUE TITLE
Scarab MacOS not opening

### ISSUE THREAD

**Issue by zombielibrary** [fifty-six/Scarab#51]
Scarab shuts down immediately upon trying to run the program.
The crash report gives the following information:
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called

**Comment by jgreever** [comment]
Current version (v1.10.0.0) is working fine for me in macOS 12.3 (21E230).

When it crashes, does it show a backtrace with a bunch of information in the window or just what you posted. This helps to diagnose what is causing the crash, as it may be something else on your system causing it.

I would also recommend creating a new user account and testing it inside that profile on your mac. [Set up users, guests, and groups on Mac - Apple Support](https://support.apple.com/guide/mac-help/set-up-other-users-on-your-mac-mtusr001/mac)

If it works fine on the new profile, you have something in your normal user account that is causing the crash. Also, have you attempted to boot into recovery mode and use Disk Utility to check your drive for errors? [Recovery Mode](https://support.apple.com/en-us/HT201314) / [Check/Repair Disk](https://support.apple.com/en-us/HT210898)

My other thought is that a mod that was installed became corrupted and might be causing the Scarab.app to crash out. As I said earlier, if it gives you a backtrace window, copy and paste the contents into a new [pastebin](https://pastebin.com/) note and post the URL to that pastebin in here (mostly because backtrace logs can be super long and can clutter the thread sometimes), but if you can't get it on pastebin, post it in here anyways that way we can see what's going on.

Thank you!

**Comment by fifty-six** [comment]
Yeah, what macOS version is it? Is this with an M1 mac or an older one? What version of Scarab is it?

**Comment by zombielibrary** [comment]
Sorry for taking so long. I did some digging and it looks likely that it's a problem with how the laptop was set up, or some archaic admin program running in the background.
I didn't ignore your advice: it didn't work on another account, my disk is fine, and I never ended up downloading any mods.
Pastebin with the translated report: https://pastebin.com/Fe4YGp2z
Pastebin with full report: https://pastebin.com/BCqxdSWK

It's a MacBook Air on the most recent OS with multiple different versions of Scarab.

**Comment by tantalor** [comment]
Same issue.

* Scarab v1.11.0.0 
* macOS 12.3.1 (21E258)
* 3 GHz 6-Core Intel Core i5

I have no `~/.config`

No crash reports or logs in Console

**Comment by jgreever** [comment]
> Same issue.
> 
>     * Scarab v1.11.0.0
> 
>     * macOS 12.3.1 (21E258)
> 
>     * 3 GHz 6-Core Intel Core i5
> 
> 
> I have no `~/.config`
> 
> No crash reports or logs in Console
> 
> I am administrator user.

If you open the Console app and search _(in the upper right corner of the window)_ for the word `run` then press enter, next click `ANY` in front of the search phrase and select `Process`, then select the `Process` button and select `Equals` at the bottom.

Once you have done this, launch Scarab again and you should see some entries for the process `run` that look similar to this (probably wont look exactly like this, but you will see the process `run` and no other processes):

![Screen Shot 2022-04-21 at 2 41 42 PM](https://user-images.githubusercontent.com/4977981/164556764-c1c7f00b-61ea-4eef-8fc3-a0c5d9a08b8b.png)

Check that for any output when attempting to launch the app and post the screenshot or text log showing any output from the `run` process. When Scarab.app is launched, it starts by launching an app called `run` inside `/Applications/Scarab.app/Contents/MacOS/run`, hence why we are looking specifically for that entry in the processes an not something like `Scarab`.

Let us know if you have any difficulties finding these log entries or posting them here.

**Comment by tantalor** [comment]
Thanks! I was able to launch Scarab by opening up the package contents (in Finder) and double clicking the `run` program. It seems to work.

However just double clicking on the Scarab app does nothing; there is no output in Console.

**Comment by fifty-six** [comment]
Reopen this if you can still repro, cleaning up old issues and I think this should be fixed.
